src/client/components/DicePanel.tsx

```diff
@@ -1,57 +1,382 @@
+          className={clsx("tab-button", active === "Skills" ? "active" : "")}
+          onClick={() => setActive("Skills")}
+        >
+          <FontAwesomeIcon icon={faHandPaper} name={"Skills"} />
+        </div>
+        {myself.diceTemplateCategories.map((category) => (
+          <DiceTemplateButton
+            key={category.id}
+            category={category}
+            active={active === category.id}
+            onSetActive={() => setActive(category.id)}
+          />
+        ))}
+        {myself.diceTemplateCategories.length < userCategoryIcons.length && (
           <div
-            key={icon.iconName}
-            className={clsx("tab-button", active === index ? "active" : "")}
-            onClick={() => setActive(index)}
+            className={"tab-button"}
+            onClick={() => {
+              addTemplateCategory();
+            }}
+            title="Add Category"
           >
-            <FontAwesomeIcon icon={icon} fixedWidth />
+            <FontAwesomeIcon icon={faPlus} fixedWidth />
           </div>
-        ))}
+        )}
       </div>
 
-      <div className={clsx("tab", active === 0 ? "active" : "")}>
-        <DiceInterface />
-        <DiceInput />
-      </div>
-      {diceCategories.map((icon, index) => (
-        <div
-          key={icon.iconName}
-          className={clsx("tab", active === index + 1 ? "active" : "")}
-        >
-          <DiceTemplates categoryIndex={index} />
-        </div>
-      ))}
+      {renderContent(active)}
     </div>
   );
 });
+
+function DiceTemplateButton({
+  category,
+  active,
+  onSetActive,
+}: {
+  category: RRDiceTemplateCategory;
+  active: boolean;
+  onSetActive: () => void;
+}) {
+  const [addMenuVisible, setAddMenuVisible] = useState(false);
+
+  return (
+    <Popover
+      interactive
+      onClickOutside={() => setAddMenuVisible(false)}
+      visible={addMenuVisible}
+      placement="bottom"
+      content={<DiceTemplateCategoryEditor category={category} />}
+    >
+      <div
+        className={clsx("tab-button", active ? "active" : "")}
+        onClick={(e) => {
+          e.stopPropagation();
+          if (e.button === 0) onSetActive();
+          else if (e.button === 2) {
+            e.stopPropagation();
+          }
+        }}
+        onContextMenu={(e) => {
+          e.stopPropagation();
+          e.preventDefault();
+          if (
+            fixedCategoryIcons.findIndex(
+              (fixedIcon) => fixedIcon === category.icon
+            ) > -1
+          )
+            return;
+          setAddMenuVisible(true);
+        }}
+        title={category.categoryName}
+      >
+        <FontAwesomeIcon icon={iconMap[category.icon]} fixedWidth />
+      </div>
+    </Popover>
+  );
+}
+
+function DiceTemplateCategoryEditor({
+  category,
+}: {
+  category: RRDiceTemplateCategory;
+}) {
+  const myself = useMyProps("diceTemplateCategories", "id");
+  const dispatch = useServerDispatch();
+
+  function deleteCategory() {
```

Comment:
Should we `useConfirm` to ask the user for confirmation before deleting a category?

Reply:
I am a bit hesitant to use pop ups, but I guess in this case it might make sense. Before people start accidentally deleting all their templates.

Reply:
I am also always in favor of a nicer UI that doesn't rely on popups 👍 To me, popups are mostly a placeholder for 'we haven't made the UX great yet'. 
It's just that I would be super sad if I accidentally deleted my dice templates^^

Reply:
So for all of our UX, yes? :smile:

Reply:
Psssssst... don't tell anyone! :D